### PR TITLE
R4 Location Read

### PR DESCRIPTION
**Pull request by HP082979**
Added location in millennium R4 entities

Description
Screen Shot 2020-08-25 at 5 26 48 PM
Screen Shot 2020-08-25 at 5 27 32 PM
Screen Shot 2020-08-25 at 5 28 12 PM
Screen Shot 2020-08-25 at 5 29 59 PM
Screen Shot 2020-08-25 at 5 32 34 PM
Screen Shot 2020-08-25 at 5 33 31 PM

----
Added Location Entity for Millennium R4 Entities
PR Checklist
----
- [X] Screenshot(s) of changes attached before changes merged.
- [ ] Screenshot(s) of changes attached after changes merged and published.

### CODE REVIEW

File: lib/resources/example_json/r4_examples_location.rb [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+        ],
+        'city': 'Bangalore',
+        'district': 'Indian River',
+        'state': 'AA',
+        'country': 'Ireland'
+      },
+      'physicalType': {
+        'coding': [
+          {
+            'system': 'https://fhir.cerner.com/2c400054-42d8-4e74-87b7-80b5bd5fde9f/codeSet/222',
```
Comment (praneethk2094) [inline lib/resources/example_json/r4_examples_location.rb:47]: Can we update the tenant to - 'ec2458f2-1e24-41c8-b71b-0e701af7583d'

File: lib/resources/example_json/r4_examples_location.rb [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
```
Comment (praneethk2094) [inline lib/resources/example_json/r4_examples_location.rb:106]: Can we update the tenant to - 'ec2458f2-1e24-41c8-b71b-0e701af7583d'

File: lib/resources/example_json/r4_examples_location.rb [hunk 3]
```diff
@@ -0,0 +1,123 @@
+# frozen_string_literal: true
+
+module Cerner
+  module Resources
+    R4_LOCATION_BUNDLE ||= {
```
Comment (praneethk2094) [inline lib/resources/example_json/r4_examples_location.rb:5]: Bundled responses have a few additional fields in comparison to a Read data such as these fields - [NutirtionOrder example](https://github.com/cerner/fhir.cerner.com/blob/main/lib/resources/example_json/r4_examples_nutrition_order.rb#L240-L249)

Can we update to have those fields? we could also do something like here: [r4_examples_nutrition_order](https://github.com/cerner/fhir.cerner.com/blob/main/lib/resources/example_json/r4_examples_nutrition_order.rb#L239)

File: content/millennium/r4/entities/location.md [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+{:toc}
+
+## Overview
+
+The Location resource describes physical places where healthcare services are provided.  In Millennium, facilities are the top level of the patient location hierarchy.  A facility is also an Organization resource at which patient locations are associated.  All facilities are organizations, but not all organizations are facilities.  A location can also be an ambulatory patient care area like a clinic or an emergency room.  The location hierarchy from highest to lowest is facility, building, nursing unit, room, and bed.
+
+
+The following fields are returned if valued:
+
+* [Location id](http://hl7.org/fhir/r4/resource-definitions.html#Resource.id){:target="_blank"}
```
Comment (BS062721) [inline content/millennium/r4/entities/location.md:17]: location id -> id?

File: content/millennium/r4/entities/location.md [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+* [Status](http://hl7.org/fhir/R4/location-definitions.html#Location.status){:target="_blank"}
+* [Name](http://hl7.org/fhir/R4/location-definitions.html#Location.name){:target="_blank"}
+* [Alias](http://hl7.org/fhir/R4/location-definitions.html#Location.alias){:target="_blank"}
+* [Mode](http://hl7.org/fhir/R4/location-definitions.html#Location.mode){:target="_blank"}
+* [Telecom](http://hl7.org/fhir/R4/location-definitions.html#Location.telecom){:target="_blank"}
+* [Address](http://hl7.org/fhir/R4/location-definitions.html#Location.address){:target="_blank"}
+* [Physical Type](http://hl7.org/fhir/R4/location-definitions.html#Location.physicalType){:target="_blank"}
+* [Managing Organization](http://hl7.org/fhir/R4/location-definitions.html#Location.managingOrganization){:target="_blank"}
+* [Part Of](http://hl7.org/fhir/R4/location-definitions.html#Location.partOf){:target="_blank"}
```
Comment (BS062721) [inline content/millennium/r4/entities/location.md:26]: Part of-> partOf

@praneethk2094 I am getting this correct? The field listed here should be in the same case they are on spec?

File: lib/resources/r4/location.yaml [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+    action: terminology
+    binding:
+      description: Physical form of the location.
+      terminology:
+        - display: Location physicalType
+          system: http://terminology.hl7.org/CodeSystem/location-physical-type
+          info_link: http://hl7.org/fhir/R4/valueset-location-physical-type.html
+        - display: Millennium Location Types
+          system: https://fhir.cerner.com/&lt;EHR source id&gt;/codeSet/222
+          info_link: http://localhost:3000/millennium/r4/proprietary-codes-and-systems/#code-set-222-location-type
```
Comment (BS062721) [inline lib/resources/r4/location.yaml:18]: This is pointing to localhost?

Comment (BS062721) [reply lib/resources/r4/location.yaml:18]: It should be: https://fhir.cerner.com/millennium/r4/proprietary-codes-and-systems/#code-set-222-location-type

File: lib/resources/example_json/r4_examples_location.rb (hunk 1 above)
Comment (HP082979) [reply lib/resources/example_json/r4_examples_location.rb]: Fix for this:
https://github.com/cerner/fhir.cerner.com/commit/533e50911cc35b3e9313e0184120b17340db3e04

File: lib/resources/example_json/r4_examples_location.rb (hunk 2 above)
Comment (HP082979) [reply lib/resources/example_json/r4_examples_location.rb]: Fix for this:
https://github.com/cerner/fhir.cerner.com/commit/533e50911cc35b3e9313e0184120b17340db3e04

File: lib/resources/example_json/r4_examples_location.rb (hunk 3 above)
Comment (HP082979) [reply lib/resources/example_json/r4_examples_location.rb:5]: Fix for this:
https://github.com/cerner/fhir.cerner.com/commit/533e50911cc35b3e9313e0184120b17340db3e04

File: content/millennium/r4/entities/location.md (hunk 4 above)
Comment (HP082979) [reply content/millennium/r4/entities/location.md:17]: This is done as per this Jira:
https://jira2.cerner.com/browse/OPENSVC-15097

File: content/millennium/r4/entities/location.md (hunk 5 above)
Comment (HP082979) [reply content/millennium/r4/entities/location.md:26]: This is done as per this Jira:
https://jira2.cerner.com/browse/OPENSVC-15097

File: lib/resources/r4/location.yaml (hunk 6 above)
Comment (HP082979) [reply lib/resources/r4/location.yaml]: Fix for this:
https://github.com/cerner/fhir.cerner.com/commit/98602c8acc06a2175351fb1d3b11577736eec395

**Comment by MaxPhilips** [comment]
@HP082979 looks like the first checkbox on this PR was rendering incorrectly due to an extra space inside the square brackets. I noticed this while checking current status of deployable PRs and noticed only 1 task was noted on this PR. I updated the description so the first task is tracked now

**Comment by lalitha94** [comment]
Closing this as these changes are made as a part of https://github.com/cerner/fhir.cerner.com/pull/520

**Pull request closed without merging**